### ISSUE TITLE
Make `directives` and `settings` required in the configuration schema

### ISSUE THREAD

**Issue by alexravenna** [microsoft/winget-cli#3500]
### Description of the new feature / enhancement

According to [the documentation](https://learn.microsoft.com/en-us/windows/package-manager/configuration/create), in a WinGet Configuration file:

> Each resource must include directives and settings.

However, this is not reflected in the [schema](https://github.com/microsoft/winget-cli/blob/master/schemas/JSON/configuration/configuration.schema.0.2.json).

### Proposed technical implementation details

Mark the two properties as `required`.

**Comment by stephengillie** [comment]
Alternatively, we could update the documentation to state that these are optional.

**Comment by JohnMcPMS** [comment]
> Alternatively, we could update the documentation to state that these are optional.

Indeed, both are optional, although there will be *very* few instances where no `settings` would be provided.  Despite that, they are not required.

**Comment by JohnMcPMS** [comment]
https://github.com/MicrosoftDocs/windows-dev-docs/pull/4560

**Comment by alexravenna** [comment]
Thanks foe the clarification and the docs PR, @JohnMcPMS!
